### PR TITLE
feat(Button): animated label

### PR DESCRIPTION
**Pull request by marcinsawicki**
Resolves: #{issue-number}

## Description
A new `animatedLabel` prop will enable showing the button label on hover/focus.

## Storybook

https://feature-button-animation--613a8e945a5665003a05113b.chromatic.com/?path=/story/components-button--button-with-animated-label

## Checklist

**Obligatory:**

- [x] Self review (use this as your final check for proposed changes before requesting the review)
- [x] Add correct label
- [ ] Assign pull request with the correct issue

### CODE REVIEW

File: packages/react-components/src/components/Button/Button.helpers.ts [hunk 1]
```diff
@@ -16,3 +18,33 @@ export const getSpinnerColors = (
...
+
+export const handleMouseInteraction = (
+  event:
+    | React.MouseEvent<HTMLButtonElement>
+    | React.MouseEvent<HTMLAnchorElement>,
+  internalHandler: () => void,
+  userHandler?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>
+): void => {
+  internalHandler();
+  userHandler && userHandler(event);
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/Button/Button.helpers.ts:30]:
```suggestion
  userHandler?.(event);
```

File: packages/react-components/src/components/Button/Button.helpers.ts [hunk 2]
```diff
@@ -16,3 +18,33 @@ export const getSpinnerColors = (
...
+
+export const handleKeyboardInteraction = (
+  event:
+    | React.FocusEvent<HTMLButtonElement>
+    | React.FocusEvent<HTMLAnchorElement>,
+  internalHandler: () => void,
+  userHandler?: React.FocusEventHandler<HTMLButtonElement | HTMLAnchorElement>
+): void => {
+  internalHandler();
+  userHandler && userHandler(event);
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/Button/Button.helpers.ts:41]:
```suggestion
  userHandler?.(event);
```

File: packages/react-components/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -62,12 +71,16 @@ export const Button = React.forwardRef<
       children,
       href,
       onClick,
+      animatedLabel,
       ...props
     },
     ref
   ) => {
+    const [labelWidth, setLabelWidth] = React.useState<number>(0);
+    const [isLabelOpen, setIsLabelOpen] = React.useState<boolean>(false);
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/Button/Button.tsx:80]: There is no need to pass types if you providing the initial values to the useState.

File: packages/react-components/src/components/Button/Button.tsx [hunk 4]
```diff
@@ -38,6 +43,10 @@ export type ButtonProps = {
    * Specify the place to render element given in `icon` prop
    */
   iconPosition?: 'left' | 'right';
+  /**
+   * Set to show label on hover if icon is present
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/Button/Button.tsx:47]: Can you please write a more descriptive explanation of the property

**Review by VadymBezpalko — APPROVED** [review]

**Review by vladko-uxds — APPROVED** [review]
Cool

**Pull request merged**